service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/NarrativeStatementMapper.java

```diff
@@ -19,19 +19,27 @@
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class NarrativeStatementMapper {
 
-    private final MessageContext messageContext;
-
     private static final Mustache NARRATIVE_STATEMENT_TEMPLATE = TemplateUtils.loadTemplate("ehr_narrative_statement_template.mustache");
 
+    private final MessageContext messageContext;
+    private final ParticipantMapper participantMapper;
+
     public String mapObservationToNarrativeStatement(Observation observation, boolean isNested) {
+        final IdMapper idMapper = messageContext.getIdMapper();
         var narrativeStatementTemplateParameters = NarrativeStatementTemplateParameters.builder()
-            .narrativeStatementId(messageContext.getIdMapper().getOrNew(ResourceType.Observation, observation.getId()))
+            .narrativeStatementId(idMapper.getOrNew(ResourceType.Observation, observation.getId()))
             .availabilityTime(getAvailabilityTime(observation))
             .comment(observation.getComment())
-            .isNested(isNested)
-            .build();
+            .isNested(isNested);
+
+        if (observation.hasPerformer()) {
+            final String participantReference = idMapper.getOrNew(observation.getPerformerFirstRep());
```

Comment:
I don't think we can use getOrNew here. The agentDirector is created before any resources are mapped. If we don't have a mapping from the resource to its Agent then we're not referring to anything here. I think a "miss" here on the idMapper would an error since it would indicate something is missing from the patient's record.

Reply:
So is the idea to have this in `IdMapper`: `public String get(Reference) throws EhrMapperException { ... }`?

Note we've already used `getOrNew` for the AC1 of this story; I can make a new PR to correct that.